components/common-go/baseserver/options.go

```diff
@@ -34,6 +35,8 @@ type config struct {
 	healthHandler healthcheck.Handler
 
 	grpcHealthCheck grpc_health_v1.HealthServer
+
+	rateLimits map[string]gitpod_grpc.RateLimit
 }
 
 func defaultConfig() *config {
```

Comment:
IMHO we should move to a common config struct. The more things we add through functional options, the more translating between config structs we'll need to do.

Reply:
I think we'll always need some level of translation. For example, to pass in a `grpc_health_v1.HealthServer` above, we need to do this during instantiation rather than during config. My thinking is that we define a YAML/JSON serializable config, which can be (auto) converted into Options which are then passed directly into the baseserver. This way, we can share standard config but retain the ability to override in code where necessary. 

There is already a reasonable amount of drift in components with respect to how they accept config and how they are configured at runtime (services etc). Firstly unifying the servers through (options) and then normalising these options into config feels like the more iterative and gradual migration